vars/tofu.groovy

```diff
@@ -0,0 +1,224 @@
+// Tofu/Terraform operations for Jenkins pipelines (Docker-based)
+
+// Get the Docker image to use for tofu commands
+def _getImage() {
+    return 'rancher-infra-tools:latest'
+}
+
+// Run a command in the tofu/ansible container
+def _runInContainer(String command, Map envVars = [:], boolean returnStdout = false) {
+    def workspace = steps.pwd()
+    
+    // Build environment variable arguments
+    // AWS credentials will be automatically inherited from withCredentials block
+    def envArgs = "-e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY"
+    if (envVars) {
+        envArgs += " " + envVars.collect { k, v -> "-e ${k}='${v}'" }.join(' ')
+    }
+    
+    def dockerCommand = "docker run --rm --platform linux/amd64 ${envArgs} -v ${workspace}:/workspace -w /workspace ${_getImage()} sh -c \"${command}\""
+    
+    if (returnStdout) {
+        return steps.sh(script: dockerCommand, returnStdout: true).trim()
+    } else {
+        return steps.sh(script: dockerCommand, returnStatus: true)
+    }
+}
+
+// Initialize Tofu backend with S3 configuration using init-backend.sh script
+// [ dir: string, bucket: string, key: string, region: string, dynamodbTable?: string ]
+def initBackend(Map config) {
+    if (!(config.dir && config.bucket && config.key && config.region)) {
+        error 'Directory, bucket, key, and region must be provided for backend initialization.'
+    }
+
+    steps.echo "Initializing Tofu backend in ${config.dir}"
+    
+    // Build the init-backend.sh command
+    def scriptArgs = [
+        "s3",
+        "--bucket ${config.bucket}",
+        "--key ${config.key}",
+        "--region ${config.region}"
+    ]
+    
+    if (config.dynamodbTable) {
+        scriptArgs.add("--dynamodb-table ${config.dynamodbTable}")
+    }
+    
+    // Run the init-backend.sh script which generates backend.tf and runs tofu init
+    def initCommand = "cd ${config.dir} && ./scripts/init-backend.sh ${scriptArgs.join(' ')}"
```

Comment:
not a fan of this being hard-coded, do we do this anywhere else in the shared lib?

what about something like this instead? https://github.com/rancher/qa-jenkins-library/blob/61a5552bd038974ca98466dbe659d44e41fb17e9/vars/container.groovy#L73-L82

Reply:
I'm not seeing the problem as it's only designed to run the init-backend script.